fluent-plugin-events/lib/sumologic/kubernetes/connector.rb

```diff
@@ -7,28 +7,43 @@ module Connector
       K8_POD_CA_CERT = 'ca.crt'.freeze
       K8_POD_TOKEN = 'token'.freeze
 
-      # Need different clients to access different API groups/versions
       CORE_API_VERSIONS = ['v1'].freeze
 
+      def connect_kubernetes
+        @clients = @api_version == 'v1' ? core_clients : core_clients.merge(group_clients)
+      end
+
       def core_clients
         CORE_API_VERSIONS.map do |ver|
           [ver, create_client('api', ver)]
         end.to_h
       end
 
-      # If @api_version is not v1, we will create client for other API groups/versions
       def group_clients
-        [@api_version].map do |ver|
-          [ver, create_client('apis', ver)]
-        end.to_h
+        elems = @api_version.split("/")
+        if elems.empty? then return {} end
+
+        # only a version: create a non-group api
+        if elems.length < 2
```

Comment:
Since you're supporting here both 'v1' and 'foo/v2' formats, do you still need separate `core_clients` vs `group_clients` methods and separate config params / consts?

Your call of course, just wanted to point out that from kubeclient's perspective, aside from Client construction, there is no difference between "core" API and other API groups...

Reply:
Thanks @cben!
I gave it a shot.